lib/src/core.dart

```diff
@@ -83,7 +83,9 @@ class AudioSession {
     _avAudioSession?.routeChangeStream
         .where((routeChange) =>
             routeChange.reason ==
-            AVAudioSessionRouteChangeReason.oldDeviceUnavailable)
+                AVAudioSessionRouteChangeReason.oldDeviceUnavailable ||
+            routeChange.reason ==
+                AVAudioSessionRouteChangeReason.newDeviceAvailable)
```

Comment:
I think this code was originally added to detect headphones being unplugged, hence line 88 below:

```dart
_becomingNoisyEventSubject.add(null);
```

So that line should be guarded so that it doesn't execute when headphones are plugged in.